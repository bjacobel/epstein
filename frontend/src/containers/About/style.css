@value margin from '../../stylesheets/measurements';
@value blue, dark-blue from '../../stylesheets/colors';

.aboutPage {
  composes: roboto from '../../stylesheets/shared.css';
  max-width: 650px;
  margin: 0 auto;

  h2,
  h3,
  h4 {
    font-weight: normal;
  }

  h3,
  h4 {
    margin-top: calc(2 * margin);
  }

  a {
    color: blue;
    text-decoration: none;

    &:hover {
      text-decoration: underline;
    }

    &:visited {
      color: dark-blue;
    }
  }
}
