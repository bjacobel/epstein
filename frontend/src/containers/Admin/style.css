@value margin from '../../stylesheets/measurements.css';

.adminConsole {
  max-width: 650px;
  margin: 0 auto;
}

.form {
  composes: roboto from '../../stylesheets/shared.css';
  width: 100%;
  display: flex;
  flex-direction: column;

  label {
    display: flex;
    margin-bottom: calc(margin / 2);
    align-items: baseline;

    span {
      flex-basis: 100px;
      text-align: right;
      margin-right: margin;
    }

    input {
      flex-grow: 1;
      padding: calc(margin / 4);
    }
  }
}

.submit {
  composes: hoverEffect from '../../stylesheets/shared';
  width: 100%;
  cursor: pointer;
  font-size: 0.8em;
  outline: none;
  display: block;
  padding: calc(margin / 4) calc(margin / 2);
  height: 35px;
  border: 1px solid lightgrey;
}

.backBtn {
}
