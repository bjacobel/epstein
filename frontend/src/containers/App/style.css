@value margin, mobile from '../../stylesheets/measurements';

:global {
  body {
    margin: 0;
    box-sizing: border-box;
  }
}

.appWrap {
  margin: 0 margin;

  @media (max-width: mobile) {
    margin: 0 calc(margin / 2);
  }
}
