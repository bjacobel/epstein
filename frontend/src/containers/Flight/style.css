@value margin from '../../stylesheets/measurements';
@value darkgrey from '../../stylesheets/colors';

.passengers {
  ul {
    margin: 0;
    padding-left: margin;
  }
}

.noslug {
  composes: robotoMono from '../../stylesheets/shared';
  color: darkgrey;
}
