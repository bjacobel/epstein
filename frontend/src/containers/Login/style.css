.loginLink {
  composes: roboto from '../../stylesheets/shared.css';
  composes: link from '../../stylesheets/shared.css';
  composes: center from '../../stylesheets/shared.css';
}
