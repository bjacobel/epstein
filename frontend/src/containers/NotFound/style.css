.not-found {
  composes: center from '../../stylesheets/shared';
  composes: roboto from '../../stylesheets/shared';
  font-weight: 300;
}
