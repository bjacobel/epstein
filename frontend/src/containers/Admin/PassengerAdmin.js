import React, { useState } from 'react';
import { useMutation, gql } from '@apollo/client';

import { AdminClient } from '../../utils/graphqlClient';
import Loading from '../../components/Loading';
import { form, submit } from './style.css';

const CREATE_OR_UPDATE_PASSENGER = gql`
  mutation(
    $slug: String!
    $name: String
    $biography: String
    $wikipedia_link: AWSURL
    $image: AWSURL
  ) {
    createOrUpdatePassenger(
      slug: $slug
      name: $name
      biography: $biography
      wikipedia_link: $wikipedia_link
      image: $image
    ) {
      id
      slug
      name
      biography
      wikipedia_link
      image
    }
  }
`;

export default (props = {}) => {
  const fields = ['slug', 'name', 'biography', 'wikipedia_link', 'image'];
  const { mode } = props;
  const filteredProps = Object.entries(props)
    .filter(([key]) => fields.includes(key))
    .reduce((prev, [k, v]) => ({ ...prev, [k]: v }), {});
  const [fieldData, setFieldData] = useState(filteredProps);

  const [
    createOrUpdatePassenger,
    { loading: mutationLoading, error: mutationError, data: mutationResult },
  ] = useMutation(CREATE_OR_UPDATE_PASSENGER, {
    variables: { ...fieldData },
    client: AdminClient,
  });

  const handleFormSubmit = event => {
    createOrUpdatePassenger();
    event.preventDefault();
  };

  if (mutationLoading) return <Loading />;
  if (mutationError) throw mutationError;

  return (
    <>
      <form className={form} onSubmit={handleFormSubmit}>
        {fields.map(field => (
          <label htmlFor={field} key={field}>
            <span>{field}</span>
            <input
              type="text"
              name={field}
              value={fieldData[field] || ''}
              onChange={ev =>
                setFieldData({ ...fieldData, [ev.target.name]: ev.target.value })
              }
            />
          </label>
        ))}
        <input className={submit} type="submit" value={`${mode} passenger`} />
      </form>
      {mutationResult && (
        <div>
          <span>mutation complete</span>
          <pre>{JSON.stringify(mutationResult, null, 2)}</pre>
        </div>
      )}
    </>
  );
};
