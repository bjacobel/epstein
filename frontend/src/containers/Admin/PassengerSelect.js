import React, { useState } from 'react';
import { gql, useQuery } from '@apollo/client';

import PassengerAdmin from './PassengerAdmin';
import LiteralsAdmin from './LiteralsAdmin';
import Loading from '../../components/Loading';
import { AdminClient } from '../../utils/graphqlClient';
import { breadcrumb, breadcrumbs, divider, selectPass } from './style.css';

export const PASSENGERS = gql`
  query {
    passengers(includeUnverified: false) {
      edges {
        ... on VerifiedPassenger {
          id
          slug
          name
          biography
          wikipedia_link
          image
          literals
        }
      }
    }
  }
`;

export default props => {
  const { clientForTests } = props || {};
  const [createFormSelected, selectCreateForm] = useState(false);
  const [currentPassengerSlug, setCurrentPassengerSlug] = useState();
  const { data, loading, error } = useQuery(PASSENGERS, {
    client: clientForTests || AdminClient,
  });

  let passengerData;
  if (data) {
    passengerData = data.passengers.edges.find(p => p.slug === currentPassengerSlug);
  }

  const handleSelectChange = ({ target }) => {
    if (target.value === 'new') return selectCreateForm(true);
    return setCurrentPassengerSlug(target.value);
  };

  if (loading) return <Loading />;
  if (error) throw error;

  const crumbs = current => (
    <div className={breadcrumbs}>
      <button
        type="button"
        className={breadcrumb}
        onClick={() => {
          selectCreateForm(false);
          setCurrentPassengerSlug();
        }}
      >
        Admin
      </button>
      <span className={divider} />
      <span className={breadcrumb}>{current}</span>
    </div>
  );

  return (
    <>
      {(createFormSelected || passengerData) &&
        crumbs(passengerData ? `Edit ${passengerData.name}` : 'Create new passenger')}
      {createFormSelected && (
        <PassengerAdmin
          mode="create"
          onCreate={slug => {
            setCurrentPassengerSlug(slug);
            selectCreateForm(false);
          }}
        />
      )}
      {passengerData && (
        <div>
          <PassengerAdmin mode="update" {...passengerData} />
          <LiteralsAdmin passenger={passengerData} />
        </div>
      )}
      {!(createFormSelected || passengerData) && (
        <label className={selectPass} htmlFor="pass-select">
          Select a passenger
          <select name="pass-select" onChange={handleSelectChange}>
            <option>-----</option>
            <option value="new">Create new passenger</option>
            {data.passengers.edges.map(p => (
              <option key={p.slug} value={p.slug}>
                {p.name}
              </option>
            ))}
          </select>
        </label>
      )}
    </>
  );
};
