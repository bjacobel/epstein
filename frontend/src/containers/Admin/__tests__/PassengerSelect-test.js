import React from 'react';
import { ApolloClient, InMemoryCache } from '@apollo/client';
import { mount } from 'enzyme';

import PassengerAdmin from '../PassengerAdmin';
import PassengerSelect, { PASSENGERS } from '../PassengerSelect';
import MockLink from '../../../utils/testing/MockLink';
import updateWrapper from '../../../utils/testing/updateWrapper';
import { breadcrumb } from '../style.css';

const passengersData = {
  data: {
    passengers: {
      edges: [
        {
          id: 1,
          slug: 'bill-clinton',
          name: 'Bill Clinton',
          biography: '',
          wikipedia_link: '',
          image: '',
        },
      ],
    },
  },
};

describe('PassengerSelect component', () => {
  let link, client;
  beforeEach(() => {
    link = new MockLink();
    client = new ApolloClient({
      cache: new InMemoryCache({
        addTypename: false,
      }),
      link,
    });
  });

  it('renders a dropdown with passengers from a query initially', async () => {
    const queryHandler = jest.fn().mockResolvedValue(passengersData);
    link.setRequestHandler(PASSENGERS, queryHandler);

    const wrapper = mount(<PassengerSelect clientForTests={client} />);

    await updateWrapper(wrapper);
    expect(wrapper.find(PassengerSelect).children()).toMatchSnapshot();
  });

  it('renders a blank PassengerAdmin if the "new passenger" option is picked', async () => {
    const queryHandler = jest.fn().mockResolvedValue(passengersData);
    link.setRequestHandler(PASSENGERS, queryHandler);

    const wrapper = mount(<PassengerSelect clientForTests={client} />);

    await updateWrapper(wrapper);
    wrapper
      .find(PassengerSelect)
      .find('select')
      .simulate('change', { target: { value: 'new' } });

    expect(wrapper.find(PassengerAdmin).props()).toEqual({
      mode: 'create',
    });
  });

  it('renders a populated PassengerAdmin if existing passenger picked', async () => {
    const queryHandler = jest.fn().mockResolvedValue(passengersData);
    link.setRequestHandler(PASSENGERS, queryHandler);

    const wrapper = mount(<PassengerSelect clientForTests={client} />);

    await updateWrapper(wrapper);
    wrapper
      .find(PassengerSelect)
      .find('select')
      .simulate('change', { target: { value: 'bill-clinton' } });

    expect(wrapper.find(PassengerAdmin).props()).toEqual({
      mode: 'update',
      ...passengersData.data.passengers.edges[0],
    });
  });

  it('has a go back button that erases state and renders initial UI', async () => {
    const queryHandler = jest.fn().mockResolvedValue(passengersData);
    link.setRequestHandler(PASSENGERS, queryHandler);

    const wrapper = mount(<PassengerSelect clientForTests={client} />);

    await updateWrapper(wrapper);
    wrapper
      .find(PassengerSelect)
      .find('select')
      .simulate('change', { target: { value: 'bill-clinton' } });

    wrapper.find(`button.${breadcrumb}`).simulate('click');
    expect(wrapper.find(PassengerSelect).children()).toMatchSnapshot();
  });
});
