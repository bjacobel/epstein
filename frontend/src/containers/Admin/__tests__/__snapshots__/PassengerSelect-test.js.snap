// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PassengerSelect component has a go back button that erases state and renders initial UI 1`] = `
<label
  className="selectPass"
  htmlFor="pass-select"
>
  Select a passenger
  <select
    name="pass-select"
    onChange={[Function]}
  >
    <option>
      -----
    </option>
    <option
      value="new"
    >
      Create new passenger
    </option>
    <option
      key="bill-clinton"
      value="bill-clinton"
    >
      Bill Clinton
    </option>
  </select>
</label>
`;

exports[`PassengerSelect component renders a dropdown with passengers from a query initially 1`] = `
<label
  className="selectPass"
  htmlFor="pass-select"
>
  Select a passenger
  <select
    name="pass-select"
    onChange={[Function]}
  >
    <option>
      -----
    </option>
    <option
      value="new"
    >
      Create new passenger
    </option>
    <option
      key="bill-clinton"
      value="bill-clinton"
    >
      Bill Clinton
    </option>
  </select>
</label>
`;
